### PR TITLE
Added fuzzer for WHERE and more cases in GROUP BY

### PR DESCRIPTION
**Pull request by qoega**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added support of WHERE clause generation to AST Fuzzer and possibility to add or remove ORDER BY and WHERE clause.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Review by nickitat — APPROVED** [review]
lgtm

**Comment by nickitat** [comment]
@qoega we need even better )

**Comment by qoega** [comment]
> @qoega we need even better )

Found that when I removed GROUP BY, grouping sets etc could remain. Actually strange that code checks only group_by_with_grouping_sets and just believes that GroupBy is set correctly

**Comment by qoega** [comment]
Fails not related
Stress test (msan) - #41548 
AST fuzzer (debug) - #42159
Stress test (ubsan) - #42162

**Comment by alexey-milovidov** [comment]
> Oct 07 16:09:28 /build/src/Client/QueryFuzzer.cpp:26:1: error: duplicate include [readability-duplicate-include,-warnings-as-errors]
Oct 07 16:09:28 #include "Parsers/IAST_fwd.h"

File: src/Client/QueryFuzzer.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include <pcg_random.hpp>
 #include <Common/assert_cast.h>
 #include <Common/typeid_cast.h>
+#include "Parsers/IAST_fwd.h"
```
Comment (qoega) [inline src/Client/QueryFuzzer.cpp:26]:
```suggestion
```

**Review by qoega — COMMENTED** [review]
Sure. It was not ready when I checked fails.

**Comment by nickitat** [comment]
AST fuzzer (msan) #41613

**Review by nickitat — APPROVED** [review]

**Pull request merged**